python/singa/autograd.py

```diff
@@ -2194,6 +2194,30 @@ def log(x):
     return Log()(x)[0]
 
 
+class Squeeze(Operation):
+    def __init__(self,axis=[]):
+        super(Squeeze, self).__init__()
+        self.axis=axis
+
+    def forward(self, x):
+        self.cache=x.shape()
+        cur = list(self.cache)
```

Comment:
```
newshape = []
for i in self.axis:
   assert self.cache[i] == 1, "the length of axis {} is {}, which should be 1".format(i, self.cache[i])
for d in self.cache:
   if d > 1:    
      newshape.append(d)
```

Reply:
sorry, I fixed the bug followed your code.